### PR TITLE
Multilabel Issue Manager for Classification

### LINKED ISSUE #812: Extend Datalab to multi-label classification datasets
Allow Datalab to find label issues when `labels` is multi-label object.

Related things to look at:

Extending Datalab to other ML tasks:
https://github.com/cleanlab/cleanlab/issues/774
https://github.com/cleanlab/cleanlab/issues/765
https://github.com/cleanlab/cleanlab/pull/796

Existing Datalab code for label issues:
https://github.com/cleanlab/cleanlab/blob/master/cleanlab/datalab/internal/issue_manager/label.py

Existing Cleanlab code for multi-label classification label issues:
https://github.com/cleanlab/cleanlab/tree/master/cleanlab/multilabel_classification

Adding new issue manager in Datalab:
https://docs.cleanlab.ai/master/cleanlab/datalab/guide/custom_issue_manager.html

### PR DESCRIPTION
**Pull request by tataganesh**
## Summary

 🎯 **Purpose**: Add Multilabel Issue Manager for Classification in Datalab. This PR includes

- Addition of a new `multilabel` task with the `label` issue type
- Issue manager for label issue type in multilabel datasets

Example Code

```python
import numpy as np
from cleanlab import Datalab
from sklearn.datasets import make_multilabel_classification as make_ml_clf
from cleanlab.internal.multilabel_utils import onehot2int

np.random.seed(10)
features = np.random.rand(7, 2)
true_y = np.array(
    [[0, 1, 0], [0, 1, 1], [1, 1, 1], [1, 0, 1], [0, 1, 1], [1, 1, 0], [1, 1, 0]]
)
pred_probs = np.full_like(true_y, fill_value=0.1, dtype=float)
pred_probs[true_y == 1] = 0.9
# Flip labels of some rows to add noise
candidate_rows = np.where(true_y.sum(axis=1) < 3)[0]
noisy_rows = np.random.choice(candidate_rows, 2, replace=False)
noisy_y = true_y.copy()
noisy_y[noisy_rows] = 1 - true_y[noisy_rows]
# Convert one-hot encoded labels to integer encoding
labels = onehot2int(noisy_y)
data = {"features": features, "labels": labels}
# Run Datalab
lab = Datalab(data, task="multilabel", label_name="labels")
lab.find_issues(pred_probs=pred_probs)
lab.report()
```

Output 
```
Audit complete. 2 issues found in the dataset.
Here is a summary of the different kinds of issues found in the data:

issue_type  num_issues
     label           2

Dataset Information: num_examples: 7, num_classes: 3


----------------------- label issues -----------------------

About this issue:
        Examples whose given label(s) are estimated to be potentially incorrect
    (e.g. due to annotation error) are flagged as having label issues.
    

Number of examples with this issue: 2
Overall dataset quality in terms of this issue: 0.6714

Examples representing most severe instances of this issue:
   is_label_issue  label_score given_label predicted_label
0            True          0.1      [0, 2]             [1]
6            True          0.1         [2]          [0, 1]
1           False          0.9      [1, 2]          [1, 2]
2           False          0.9   [0, 1, 2]       [0, 1, 2]
3           False          0.9      [0, 2]          [0, 2]
```

## Impact

🌐 Areas Affected: Tasks supported by Datalab.

Testing

 🔍 Testing Done: Issue Manager tests and End-to-End Datalab tests.

## Links to Relevant Issues or Conversations

Resolves #812

### CODE REVIEW

**Comment by elisno** [comment]
Link-check failing is unrelated to the PR.

The DeadlineExceeded error in the property-test is not reproduced. It's a flaky test that isn't affected by this PR.

**Review by elisno — APPROVED** [review]
LGTM. Great work @tataganesh!
